client/state/data-layer/wpcom/comments/index.js

```diff
@@ -156,6 +158,74 @@ export const announceFailure = ( { dispatch, getState }, { siteId, postId } ) =>
 	dispatch( errorNotice( error ) );
 };
 
+// @see https://developer.wordpress.com/docs/api/1.1/post/sites/%24site/comments/%24comment_ID/delete/
+export const deleteComment = ( { dispatch, getState }, action ) => {
+	const { siteId, postId, commentId } = action;
+
+	if ( startsWith( commentId, 'placeholder' ) ) {
+		return;
+	}
+
+	const comment = getSiteComment( getState(), siteId, commentId );
+
+	if ( 'trash' !== comment.status && 'spam' !== comment.status ) {
+		dispatch( changeCommentStatus( siteId, postId, commentId, 'trash' ) );
+		return;
+	}
+
+	dispatch(
+		http(
+			{
+				method: 'POST',
+				apiVersion: '1.1',
+				path: `/sites/${ siteId }/comments/${ commentId }/delete`,
+			},
+			{
+				...action,
+				comment,
+			}
+		)
+	);
+};
+
+export const announceDeleteSuccess = ( { dispatch } ) => {
+	dispatch(
+		createNotice(
+			'is-error',
+			translate( 'Comment deleted permanently.' ),
+			{
+				duration: 5000,
+				isPersistent: true,
+			}
+		)
+	);
+};
+
+export const announceDeleteFailure = ( { dispatch, getState }, action ) => {
+	const { siteId, postId, comment } = action;
+
+	dispatch(
+		errorNotice(
+			translate( 'Could not delete the comment.' ),
```

Comment:
Hi! I've found a possible matching string that has already been translated **9** times:
`translate( 'Sorry, you can not delete this comment.' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).